File: examples/documents/ros/deploy-2048/go/demo_test.go
package main

import (
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestNewdemoStack(t *testing.T) {
	app := alicloudroscdkcore.NewApp(nil)
	props := &demoStackProps{}
	stack := NewdemoStack(app, "myTestStack", props)

	artifact := app.Synth(nil).GetStackArtifact(stack.ArtifactId()).Template()
	expect := map[string]interface{}{
		"Metadata": map[string]interface{}{
			"ALIYUN::ROS::Interface": map[string]interface{}{
				"TemplateTags": []interface{}{"Create by ROS CDK"},
			},
		},
		"ROSTemplateFormatVersion": "2015-09-01",
	}
	assert.Equal(t, expect, artifact)
}

File: examples/documents/ros/deploy-2048/csharp/src/DemoTest/DemoTest.cs
using AlibabaCloud.SDK.ROS.CDK.Core;
using NUnit.Framework;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using Demo;

namespace Stack.UnitTests.Services
{
    [TestFixture]
    public class Stack_IsStackShould
    {
        [Test]
        public void DemoStack_IsStackShould()
        {
            var app = new App();
            var testStack = new DemoStack(app, "TestStack");
            var result = app.Synth().GetStackArtifact(testStack.ArtifactId).Template;
            var actualJson = JsonConvert.SerializeObject(result);
            JObject obj = new JObject();
            JObject metadata = new JObject();
            JObject rosInterface = new JObject();
            JArray templateTags = new JArray();
            metadata.Add("ALIYUN::ROS::Interface", rosInterface);
            rosInterface.Add("TemplateTags", templateTags);
            templateTags.Add("Create by ROS CDK");
            obj.Add("Metadata", metadata);
            obj.Add("ROSTemplateFormatVersion", "2015-09-01");
            var expected = JsonConvert.SerializeObject(obj);
            Assert.AreEqual(actualJson, expected);
        }
    }
}
